src/aspire/estimation/covar2d.py

```diff
@@ -148,19 +229,20 @@ def apply(A, x):
     def shrink_covar_backward(self, b, b_noise, n, noise_var, shrinker):
         """
         Apply the shrinking method to the 2D covariance of coefficients.
+
         :param b: An input coefficient covariance.
         :param b_noise: The noise covariance.
         :param noise_var: The estimated variance of noise.
         :param shrinker: The shrinking method.
         :return: The shrinked 2D covariance coefficients.
         """
         b_out = b
-        for ell in range(0, b.size()):
+        for ell in range(0, len(b)):
             b_ell = b[ell]
             p = np.size(b_ell, 1)
             S = sqrtm(b_noise[ell])
             # from Matlab b_ell = S \ b_ell /S
-            b_ell = np.divide(solve(S, b_ell), S)
+            b_ell = solve(S, b_ell) @ inv(S)
```

Comment:
This is not very efficient (or stable). Doing something like `solve(S.T, solve(S, b_ell).T).T` is the right way to do this.